packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -250,6 +265,41 @@ module.exports = {
     ],
   },
   plugins: [
+    // configuration for vendor splitting and long term caching
+    // more info: https://medium.com/webpack/predictable-long-term-caching-with-webpack-d3eee1d3fa31
+    new webpack.NamedModulesPlugin(),
+    new webpack.NamedChunksPlugin(chunk => {
+      if (chunk.name) {
+        return chunk.name;
+      }
+      return chunk.modules
+        .map(m => path.relative(m.context, m.request))
+        .join('_');
```

Comment:
the only thing to keep in mind here is, that this could potentially become a length string, so it might make sense to cut the string at a certain point or use a hash digest or something similar.
Would probably need real life testing to see if it really becomes an issue. 

the most brutal way would be to `digest` the paths, but that would mean one looses the ability to see what actually got required. maybe a healthy mix of both?
```
new webpack.NamedChunksPlugin(chunk => {
    if (chunk.name) {
        return chunk.name;
    }
    // const crypto = require('crypto');
    const pathDigest = crypto.createHash('md5');
    return chunk.modules
        .map(m => path.relative(m.context, m.request))
        .reduce((d, m )=> d.update(m), pathDigest)
        .digest('base64');
```